C4/Circulation.pm

```diff
@@ -726,6 +726,13 @@ sub CanBookBeIssued {
     }
 	return ( \%issuingimpossible, \%needsconfirmation ) if %issuingimpossible;
 
+    if ( GetTransfers($item->{itemnumber}) ) {
+                 $issuingimpossible{RESERVED} = 1;
```

Comment:
Item can be on transfer even if there is no reserve so maybe add a new issuingimpossible value "ON_TRANSFER". Remember to add the documentation also for this value and handle this case in the parts of code where CanBookBeIssued() is called.